nonlinear_solid_mechanics/nonlinear_solid_mechanics.cc

```diff
@@ -0,0 +1,1495 @@
+/* ---------------------------------------------------------------------
+ * Copyright (c) 2020 by the preCICE authors
+ *
+ * This file is part of the dealii-adapter for the coupling library
+ * preCICE. Parts of this program are based on deal.II tutorial programs.
+ *
+ * This adapter is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version. The full text of the license can
+ * be found in the file LICENSE in the precice/dealii-adapter repository.
+ * Parts of this program are based on the previous work of Jean-Paul Pelteret
+ * and Andrew McBride in their deal.II code gallery program 'Quasi-Static
+ * Finite-Strain Compressible Elasticity'
+ * ---------------------------------------------------------------------
+ *
+ * Author: David Schneider 2020
+ */
```

Comment:
In the repo we specified LGPL, GNU Lesser General Public License. I think LGPL is good enough here. 
Normally we don't do such license statements. Reference to the original work and to you is good, but the license is anyway part of the repository.
In my opinion, more important is to have this in the README.md.
 @MakisH what do you think?

Reply:
Indeed, we currently only (normally) mention such license statements in one place in the whole repository (the `LICENSE.md` file), although we have not enforced this strictly.

I have no clue about the legal aspects, but I never really liked such headers, as they duplicate information. I would just remove it and keep everything in the README (previous work/context)  and LICENSE.

Reply:
This is a deal.II standard. You will find this lines in every deal.II tutorial file, so it is a bit of a conceptual question, whether you like to stick to a deal or a preCICE standard.

Reply:
Just discussed with @DavidSCN:
* we actually need to change from LGPL to GPL since we copy from tutorials which come under GPL.
* we will remove the license statement in the files and instead mention previous work in the README

Reply:
Just took a closer look at the licenses and I copied probably from the wrong place. The tutorials come with LGPL 2.1:

https://github.com/dealii/code-gallery/blob/7f274b2406d4f2c74adbfb2972135276be62341b/Quasi_static_Finite_strain_Compressible_Elasticity/cook_membrane.cc#L10

Therefore, the licence file should be ok, right?
I removed the initial statement now and added an entry in the README

Reply:
> Therefore, the license file should be ok, right?

yes

> I removed the initial statement now and added an entry in the README

:+1: